### PR TITLE
 Twitter Support Reimplementation  in Swift

### LINKED ISSUE #32: Reimplement Twitter integration in Swift

### PR DESCRIPTION
**Pull request by lennet**
This is a reimplementation of the Twitter integration with using the critical maps api instead of a direct connection to twitter. I also made deleted some unused icons and made some changes to the pod file so that the project is finally back to 0 warnings 🎉

bildschirmfoto 2019-01-28 um 22 53 12

We should maybe consider also showing the date of a tweet next to the hour and minute.

fixes #32

### CODE REVIEW

**Comment by zutrinken** [comment]
Here is an updated screen (please ignore NavBar and missing TabBar) with relative dates and username. It's similar to the official iOS twitter app. Optional highlighting hashtags (not sure if it makes any sense in our case).

File: CriticalMass/TwitterManager.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        formatter.dateFormat = "EEE MMM dd HH:mm:ss Z yyyy"
+        return formatter
+    }()
+
+    private let networkLayer: NetworkLayer
+
+    var updateTweetsCallback: (([Tweet]) -> Void)?
+
+    init(networkLayer: NetworkLayer, url: URL = URL(string: "https://api.criticalmaps.net/twitter")!) {
```
[the remark this answers is not in the document]
Comment (lennet) [reply CriticalMass/TwitterManager.swift]: Good point! I added a Constants file

**Comment by lennet** [comment]
> Here is an updated screen (please ignore NavBar and missing TabBar) with relative dates and username. It's similar to the official iOS twitter app. Optional highlighting hashtags (not sure if it makes any sense in our case).
> 
> 


 I also added highlighting for hashtags and usernames but they aren't interactive though for now. We could link them to the Twitter app/ Website in the future

**Comment by zutrinken** [comment]
> We could link them to the Twitter app/ Website in the future

That's a good idea. But I guess it's better not highlighting them unless they're interactive. Else it's misleading.

One more thing: Regular URLs. They should be linked as well then.

**Comment by lennet** [comment]
> > We could link them to the Twitter app/ Website in the future
> 
> That's a good idea. But I guess it's better not highlighting them unless they're interactive. Else it's misleading.
> 
> One more thing: Regular URLs. They should be linked as well then.

makes sense! I removed the highlighting and added an Issue for that feature

**Pull request merged**